### ISSUE TITLE
System.time Undervalues Time Used by read_csv

### ISSUE THREAD

**Issue by DarioS** [tidyverse/readr#719]
I read in a Gzipped CSV file with read_csv and the reported timing was

```
|========================================================================| 100%  921 MB
   user  system elapsed 
 18.008   0.000  17.902 
```

But, looking at my wristwatch, I noticed that it took 52 seconds, much closer to the 81 seconds taken by the base `read.csv` function. Is the performace difference between them overstated because of incorrect timing of `read_csv` by `system.time`?

**Comment by jimhester** [comment]
On my machine `system.time()` elapsed time matches almost exactly with wallclock time for this.

It is hard to know what is going on with your system without knowing what system you are running this on, what code you are running and what file.

But regardless I don't think there is any overstatement of the performance difference with `read.csv`.

**Comment by kevinushey** [comment]
@DarioS: what environment are you running your `read_csv()` code in? (R console? RStudio? Notebook? Other?) Does it make a difference if you disable progress bar reporting?

**Comment by DarioS** [comment]
I used RStudio yesterday. Today, I tried standalone R and it matches the watch time. I also tried different `progress` settings in RStudio. The progress bar causes a difference in timing between my watch and R and the difference to `read.csv` is much larger than I observed yesterday. For standalone R, having a progress bar or not makes no difference to timing; only in RStudio is it different. I'll remember to set it to `FALSE` for future analyses.
